### DISCUSSION TITLE
🛠️ Pydantic Settings for AWS Secrets Manager & Parameter Store

### DISCUSSION THREAD

**Discussion by ceb10n** [pydantic/pydantic#10002] [Projects using Pydantic]
I created a simple `pydantic-settings` extension to load configurations from aws secrets manager and parameter store 🤗

It can be as simple as:

```python
class ParameterStoreSettings(ParameterStoreBaseSettings):
    mongodb_host: str
    mongodb_db_name: str
```

If anyone wants to take a look:

* https://github.com/ceb10n/pydantic-settings-aws
* https://ceb10n.github.io/pydantic-settings-aws/

**Comment by keithfma** [comment]
I wonder if this could be merged into the main `pydantic-settings` package? There is already an analagous `AzureKeyVaultSettings` class, so it seems natural to me.

**Reply by samuelhwilliams** [reply]
^ this does feel like it'd be a nice thing to pull in and make officially supported

**Reply by hramezani** [reply]
pydantic-settings support azure key vault and there is a PR for adding AWS secret manager support
https://github.com/pydantic/pydantic-settings/pull/532
